apps/next/components/Register/index.tsx

```diff
@@ -197,43 +211,48 @@ export default function Register({ result }: { result: string }) {
           <div className="flex-col bg-gray-800 px-8 py-12 rounded-b-lg">
             <Alert
               available={
-                isNormalDomain && registerState !== RegisterState.Registered
+                isNormalDomain &&
+                registerState !== RegisterState.Registered &&
+                registerState !== undefined
```

Comment:
Why aren't we using an enum value to determine whether the domain is available? Why make the `registerState` variable `undefined` by default instead of setting it to a specific enum value like `Unregistered` to begin with or even `Registered` to begin with? Feels like we could reduce the number of boolean expressions throughout this component.

Reply:
If the ```registerState``` is set to either ```Unregistered``` or ```Registered```, then when the component is mounting, the page may briefly display an incorrect result for a fraction of a second. For example, if the actual domain availability is true but the default value of registerState in the code is ```Unregistered```, the page will briefly show that the domain is ```Unregistered``` and then, after the actual contract read, the page will show the correct state.

To avoid displaying an inaccurate state, even for a fraction of a second, I decided to use a skeleton animation. For this purpose, I needed to set the default value of registerState as follows: ```const [registerState, setRegisterState] = useState<RegisterState | undefined>();```

So, in short, if we allow the ```registerState``` to be ```undefined```, we can display a loading state while waiting for the correct value. Then, we can show whether the domain is registered or not once the correct value is obtained.

That being said, I actually think we can handle boolean expressions better by removing ```registerState !== undefined``` as it is unnecessary.

Reply:
Thank you for the clarification and examples Matteo, apologies if I came off ignorant or arrogant about the process. I now understand the use of `undefined` in the context of unmounted components and skeleton animations.